app/containers/Settings/Settings.jsx

```diff
@@ -67,36 +67,42 @@ type State = {
   selectedLanguage: Language,
 }
 
-export const loadWalletRecovery = (
+export const loadWalletRecovery = async (
   showSuccessNotification: Object => any,
   showErrorNotification: Object => any,
   setAccounts: (Array<Object>) => any,
 ) => {
-  dialog.showOpenDialog(fileNames => {
-    // fileNames is an array that contains all the selected
-    if (!fileNames) {
+  const { canceled, filePaths } = await dialog.showOpenDialog()
+  if (canceled || !filePaths) return
+
+  const filepath = filePaths[0]
+  fs.readFile(filepath, 'utf-8', (err, data) => {
+    if (err) {
+      showErrorNotification({
+        message: `An error occurred reading the file: ${err.message}`,
+      })
       return
     }
-    const filepath = fileNames[0]
-    fs.readFile(filepath, 'utf-8', (err, data) => {
-      if (err) {
+    const walletData = JSON.parse(data)
+
+    recoverWallet(walletData)
+      .then(data => {
+        showSuccessNotification({ message: 'Recovery was successful.' })
+        setAccounts(data.accounts)
+      })
+      .catch(err => {
         showErrorNotification({
-          message: `An error occurred reading the file: ${err.message}`,
+          message: `An error occurred recovering wallet: ${err.message}`,
         })
-        return
-      }
-      const walletData = JSON.parse(data)
+      })
+  })
+}
```

Comment:
Tiny refactor suggestion for readability:

```javascript

export const loadWalletRecovery = async (
  showSuccessNotification: Object => any,
  showErrorNotification: Object => any,
  setAccounts: (Array<Object>) => any,
) => {
  const { canceled, filePaths } = await dialog.showOpenDialog()
  if (canceled || !filePaths) return

  const filepath = filePaths[0]
  fs.readFile(filepath, 'utf-8', async (err, data) => {
    if (err) {
      showErrorNotification({
        message: `An error occurred reading the file: ${err.message}`,
      })
      return
    }
    const walletData = JSON.parse(data)
    const recoveryData = await recoverWallet(walletData).catch(err => {
      showErrorNotification({
        message: `An error occurred recovering wallet: ${err.message}`,
      })
    })

    if (recoveryData) {
      showSuccessNotification({ message: 'Recovery was successful.' })
      setAccounts(recoveryData.accounts)
    }
  })
}
```

Reply:
Can you make it a code suggestion, so I can just accept/commit? It is the button left of `H B I` in the editor. Right now I'm trying to eye ball the differences

Reply:
Somehow Github failed to replace the old code with your new suggestion and instead inserted it. I added a commit that fixed it. Should be good to go now